tools/cldr-code/src/main/java/org/unicode/cldr/util/personname/PersonNameFormatter.java

```diff
@@ -773,6 +793,21 @@ private String coalesceLiterals(StringBuilder l1, StringBuilder l2) {
             return result;
         }
 
+        private boolean endsWith(StringBuilder l1, StringBuilder l2) {
+            final int l2Length = l2.length();
+            final int delta = l1.length() - l2Length;
+            if (delta < 0) {
+                return false;
+            }
+            for (int i = 0; i < l2Length; ++i) {
+                // don't have to worry about unpaired surrogates.
+                if (l1.charAt(i+delta) != l2.charAt(i)) {
+                    return false;
+                }
+            }
+            return true;
```

Comment:
I'm a little curious about A.endsWith(B) as the algorithm you're using here.  So we'd only coalesce if A and B are equal or B is a substring of A.  Intuitively, it seems like you'd either want to require them to be exactly equal, or you'd want to look for substrings of _any_ (reasonable) length that end A and begin B.  Why the middle-ground approach here?  It seems like you're accounting for a corner case here I'm not seeing.

My first thought was that it had to do with whitespace, but you're not actually changing any of the whitespace-handling logic in this function.  Is it whitespace anyway, or are you accounting for some other situation?